### DISCUSSION TITLE
How to handle pressed button during boot

### DISCUSSION THREAD

**Discussion by zeppelin37** [bxparks/AceButton#92] [Q&A]
Hi,

I am using a button to wake my ESP32 from deep sleep. I would like to use AceButton to detect a LongPress. I read in the code that firing the event handler is on purpose avoided in this scenario. But that is exactly what I want here.
I tried to fake the "released" button state in the setup method before calling button.check(), but could not get it to work.

Do you have an idea how to implement this scenario?

Many thanks

**Comment by bxparks** [comment]
Sounds like `AceButton::isPressedRaw()` could work. See also the discussion here:
https://github.com/bxparks/AceButton/issues/21

**Reply by bxparks** [reply]
If you want to detect a LongPress after boot, you'll have to implement the timing delay and trigger yourself though.
